### ISSUE TITLE
Fcitx5 doesn't work if Telegram is launched through LXQt runner or LXQt panel's application menu

### ISSUE THREAD

**Issue by glu8716** [telegramdesktop/tdesktop#26891]
### Steps to reproduce

1. Open telegram-desktop through LXQt runner or through LXQt panel's application menu plugin
2. Try to switch between input methods with fcitx5

### Expected behaviour

Fcitx5 should work and switch input methods.

### Actual behaviour

Fcitx5 doesn't work and you can't switch between input methods. As [stated](https://github.com/lxqt/lxqt-runner/issues/265#issuecomment-1740674023) by one of the main developer of LXQt this could be a problem with Telegram ignoring Qt plugins. This has been happening for about two months. Fcitx5 still works if the program is launched from the terminal or from the desktop file through PCManFM-Qt. It doesn't when started through LXQt runner or through LXQt panel's application menu plugin as said above. 

### Operating system

Artix Linux

### Version of Telegram Desktop

4.10.1

### Installation source

Other (unofficial) source

**Comment by ilya-fedin** [comment]
Probably you have a problem with D-Bus environment. Two months is exactly the time when Telegram became D-Bus activatable.

**Comment by ilya-fedin** [comment]
Personally I recommend to use dbus-broker if you have systemd. It launches services via systemd and so `systemd --user` environment is inherited.

**Comment by glu8716** [comment]
I've never touched anything about D-Bus and I don't have a system with systemd, so I can't try what you said. Both the terminal and lxqt-runner are using the same `DBUS_SESSION_BUS_ADDRESS` environment variable, so I still don't understand why Fcitx5 works when the program is launcher through terminal, but doesn't when it's launched through lxqt-runner. Telegram is the only program that is giving me this issue. On every other program launched through lxqt-runner I can use Fcitx5 without any problems.

**Comment by ilya-fedin** [comment]
When you launch Telegram via terminal, it gets launched just as a child process. When you launch Telegram via this thing named lxqt-runner, it likely reads DBusActivatable parameter in the .desktop file and does a D-Bus request instead of launching a process. D-Bus daemon gets the request and launches Telegram. If the D-Bus daemon's environment is not properly updated by your desktop environment or by you, you will get any kind of weird issues cause all the processes started by D-Bus daemon (including Telegram) inherit its environment.

> On every other program launched through lxqt-runner I can use Fcitx5 without any problems.

This only means your software set is not large enough to include other D-Bus activatable applications.

**Comment by glu8716** [comment]
> f the D-Bus daemon's environment is not properly updated by your desktop environment or by you

I have near to zero knowledge when it comes to D-Bus. How can I check if it's updated or not?

**Comment by ilya-fedin** [comment]
I don't know. You can update it with `dbus-update-activation-environment`

**Comment by glu8716** [comment]
That didn't work for me. However after some experiments I've found out that removing `DBusActivatable=true` from the .desktop file is the solution. This might be a dumb question, but will I miss something without that?

**Comment by ilya-fedin** [comment]
> This might be a dumb question, but will I miss something without that?

Yes, you will miss D-Bus activation and all further updates to the .desktop file what will lead to malfunction.

> That didn't work for me.

I hope you ran it without sudo? It's also possible that a part of your system runs with one dbus-daemon instance while another part with another. This was quite a widespread problem before systemd has unified that and converted session bus into user bus.

**Comment by glu8716** [comment]
`I hope you ran it without sudo?`

Yes, without sudo. What I can see is that my dbus socket is located in the /tmp folder (this is what Artix does by default apparently). There are actually two dbus socket, one owned by SDDM and one owned by my user. I tried to uninstall SDDM and started X with the .xinitrc file to see if the problem was SDDM, but it wasn't. I've also tried to switch the init system (was using s6 and switched to runit), but no luck there either. The problem could be elogind... maybe I should ask on the Artix forums.

**Comment by ilya-fedin** [comment]
Ah, maybe you need to run with `--all`

**Comment by glu8716** [comment]
You are right, `dbus-update-activation-environment --all` works! I guess I'll have to run it every time I start a new session? Or is there a configuration somewhere?

**Comment by ilya-fedin** [comment]
Your DE should have some configuration for autostart scripts. Plasma does.

**Comment by glu8716** [comment]
Yes, it has. And that's exactly what I did. What I mean is, do you know what could be the component of the system that should take care of this instead of manually running the command? So that I can report this to the Artix team if needed.

**Comment by ilya-fedin** [comment]
In Plasma this is done by startplasma

**Comment by glu8716** [comment]
I see, so this is related to the DE. I'll try to reach the LXQt developers since this is my DE. Thanks for everything!

**Comment by ilya-fedin** [comment]
Note: lxqt-runner's behavior is the right one. While PCManFM-Qt perhaps has no D-Bus activation support (should work with g_app_info_launch_default_for_uri and KIO APIs which are mimeapps spec implementations but may not work with QDesktopServices::openUrl which just calls xdg-open which is a bash script with DE-dependent behavior).
